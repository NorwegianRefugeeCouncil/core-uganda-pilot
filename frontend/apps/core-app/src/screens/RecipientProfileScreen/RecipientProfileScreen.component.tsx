import * as React from 'react';
import { ScrollView, Skeleton, Text } from 'native-base';
import { Recipient } from 'core-api-client/src/types/client/Recipient';
import { FormWithRecord } from 'core-api-client';

import { RecipientViewer } from '../../components/Recipient/RecipientViewer';

import * as Styles from './RecipientProfileScreen.styles';

type Props = {
  isLoading: boolean;
  data: FormWithRecord<Recipient>[];
  error: string | null;
};

export const RecipientProfileScreenComponent: React.FC<Props> = ({
  data,
  isLoading,
  error,
}) => (
  <ScrollView width="100%">
    <Styles.Container width="100%" maxWidth="1180px" marginX="auto">
      {isLoading && !error && <Skeleton h="20" p="4" />}
      {error && (
        <Text variant="body" color="signalDanger">
          {error}
        </Text>
      )}
      {!isLoading && !error && <RecipientViewer data={data} />}
    </Styles.Container>
  </ScrollView>
);
