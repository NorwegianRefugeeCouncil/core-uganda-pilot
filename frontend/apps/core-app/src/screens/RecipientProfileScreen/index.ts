export { RecipientProfileScreenContainer as RecipientProfileScreen } from './RecipientProfileScreen.container';
