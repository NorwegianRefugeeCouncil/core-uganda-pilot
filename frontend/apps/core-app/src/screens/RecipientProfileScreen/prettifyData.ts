import { FormWithRecord } from 'core-api-client';
import { Recipient } from 'core-api-client/src/types/client/Recipient';

export const prettifyData = (
  originalData: FormWithRecord<Recipient>[],
): FormWithRecord<Recipient>[] => {
  // don't show key fields
  const dataWithoutKeys = originalData.map((ancestor) => {
    const noKeys = ancestor.form.fields.filter((field) => !field.key);

    return {
      ...ancestor,
      form: {
        ...ancestor.form,
        fields: noKeys,
      },
    };
  });

  // merge fields of individual into individual beneficiary form
  if (dataWithoutKeys.length > 1) {
    const mergedIndividual = [
      {
        form: {
          ...dataWithoutKeys[1].form,
          fields: [
            ...dataWithoutKeys[0].form.fields,
            ...dataWithoutKeys[1].form.fields,
          ],
        },
        record: {
          ...dataWithoutKeys[1].record,
          values: [
            ...dataWithoutKeys[0].record.values,
            ...dataWithoutKeys[1].record.values,
          ],
        },
      },
    ];
    return mergedIndividual.concat(dataWithoutKeys.slice(2));
  }
  return dataWithoutKeys;
};
