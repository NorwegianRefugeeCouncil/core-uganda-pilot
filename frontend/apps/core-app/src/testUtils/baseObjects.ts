import { FormDefinition, FormType, Record } from 'core-api-client';

export const baseForm: FormDefinition = {
  id: 'form1',
  code: '',
  name: 'form 1',
  databaseId: 'databaseId1',
  folderId: 'folderId1',
  formType: FormType.DefaultFormType,
  fields: [],
};

export const baseTextField = {
  id: 'field1',
  name: 'field 1',
  description: 'description 1',
  code: '',
  required: false,
  key: false,
  fieldType: {
    text: {},
  },
};

export const baseMultilineTextField = {
  id: 'field2',
  name: 'field 2',
  description: 'description 2',
  code: '',
  required: true,
  key: true,
  fieldType: {
    multilineText: {},
  },
};

export const baseRecord1: Record = {
  id: 'record1',
  values: [
    {
      value: 'text',
      fieldId: 'field1',
    },
    {
      value: 'multi line text',
      fieldId: 'field2',
    },
  ],
  formId: 'form1',
  databaseId: 'databaseId1',
  ownerId: undefined,
};
