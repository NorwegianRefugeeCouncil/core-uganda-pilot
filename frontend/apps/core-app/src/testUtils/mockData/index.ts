export { makeRecord } from './Record';
export { makeForm } from './Form';
export { makeField } from './FormField';
