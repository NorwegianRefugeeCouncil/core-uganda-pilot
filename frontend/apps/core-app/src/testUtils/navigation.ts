export const mockNavigationProp = {
  dispatch: jest.fn(),
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  getParent: jest.fn(),
  getState: jest.fn(),
  setParams: jest.fn(),
  setOptions: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  replace: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
};
