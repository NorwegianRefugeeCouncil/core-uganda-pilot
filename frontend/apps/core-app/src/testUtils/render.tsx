import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import {
  render as r,
  RenderAPI,
  RenderOptions,
} from '@testing-library/react-native';
import { render as rWeb } from '@testing-library/react';
import { NativeBaseProvider } from 'native-base';
import { theme } from 'core-design-system';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

export const render = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
): RenderAPI =>
  r(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );

export const renderWeb = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
) =>
  rWeb(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );
