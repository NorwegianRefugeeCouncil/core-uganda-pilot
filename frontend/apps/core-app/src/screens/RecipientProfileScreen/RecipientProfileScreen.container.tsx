import * as React from 'react';
import { RouteProp, useRoute } from '@react-navigation/native';
import { Recipient } from 'core-api-client/src/types/client/Recipient';
import { FormWithRecord } from 'core-api-client';

import { RootParamList } from '../../navigation/types';
import { formsClient } from '../../clients/formsClient';
import config from '../../config';

import { prettifyData } from './prettifyData';
import { RecipientProfileScreenComponent } from './RecipientProfileScreen.component';

export const RecipientProfileScreenContainer: React.FC = () => {
  const route = useRoute<RouteProp<RootParamList, 'RecipientProfile'>>();
  const [isLoading, setIsLoading] = React.useState<boolean>(true);
  const [data, setData] = React.useState<FormWithRecord<Recipient>[]>([]);
  const [prettifiedData, setPrettifiedData] = React.useState<
    FormWithRecord<Recipient>[]
  >([]);
  const [error, setError] = React.useState<string>();

  React.useEffect(() => {
    (async () => {
      try {
        const recipientData = await formsClient.Recipient.get({
          recordId: route.params.id,
          formId: config.recipient.registrationForm.formId,
          databaseId: config.recipient.registrationForm.databaseId,
        });
        setData(recipientData);
      } catch (err) {
        setError(JSON.stringify(err));
      }
      setIsLoading(false);
    })();
  }, [
    config.recipient.registrationForm.formId,
    config.recipient.registrationForm.databaseId,
    route.params.id,
  ]);

  React.useEffect(() => {
    if (data.length) {
      setPrettifiedData(prettifyData(data));
    }
  }, [data]);

  return (
    <RecipientProfileScreenComponent
      data={prettifiedData}
      isLoading={isLoading}
      error={error}
    />
  );
};
