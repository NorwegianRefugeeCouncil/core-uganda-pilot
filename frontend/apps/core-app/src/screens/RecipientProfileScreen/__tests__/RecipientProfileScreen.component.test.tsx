import { FormType } from 'core-api-client';

import { render } from '../../../testUtils/render';
import { RecipientProfileScreenComponent } from '../RecipientProfileScreen.component';

const data = [
  {
    form: {
      id: 'id',
      databaseId: 'databaseId',
      formType: FormType.RecipientFormType,
      name: 'name',
      code: 'code',
      folderId: 'folderId',
      fields: [],
    },
    record: {
      id: 'id',
      formId: 'formId',
      databaseId: 'databaseId',
      values: [],
      ownerId: undefined,
    },
  },
];

it('should match the snapshot', () => {
  const { toJSON } = render(
    <RecipientProfileScreenComponent data={data} isLoading={false} />,
  );
  expect(toJSON()).toMatchSnapshot();
});
it('should match the snapshot while loading', () => {
  const { toJSON } = render(
    <RecipientProfileScreenComponent data={data} isLoading />,
  );
  expect(toJSON()).toMatchSnapshot();
});
it('should match the snapshot with error', () => {
  const { toJSON } = render(
    <RecipientProfileScreenComponent
      data={data}
      isLoading={false}
      error="error"
    />,
  );
  expect(toJSON()).toMatchSnapshot();
});
