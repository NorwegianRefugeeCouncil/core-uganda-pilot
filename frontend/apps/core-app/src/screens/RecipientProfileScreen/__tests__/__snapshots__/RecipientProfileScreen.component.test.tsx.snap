// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {},
        Object {
          "dataSet": Object {},
        },
      ]
    }
    dataSet={Object {}}
    style={Object {}}
  >
    <View>
      <View
        dataSet={Object {}}
        style={
          Array [
            Object {},
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "paddingBottom": 8,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 8,
            }
          }
        >
          <View
            _text={
              Object {
                "color": "lightText",
              }
            }
            accessible={true}
            collapsable={false}
            dataSet={Object {}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onHoverIn={[Function]}
            onHoverOut={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#24303E",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          >
            <Text
              dataSet={Object {}}
              level="1"
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#FFFFFF",
                  "fontFamily": "Roboto_400Regular",
                  "fontSize": 18,
                  "fontStyle": undefined,
                  "fontWeight": undefined,
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "textAlign": "auto",
                  "textDecorationLine": undefined,
                }
              }
            >
              name
            </Text>
            <Text
              dataSet={Object {}}
              level="1"
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#FFFFFF",
                  "fontFamily": "Roboto_400Regular",
                  "fontSize": 18,
                  "fontStyle": undefined,
                  "fontWeight": undefined,
                  "letterSpacing": 0,
                  "lineHeight": 26,
                  "textAlign": "auto",
                  "textDecorationLine": undefined,
                }
              }
            >
              closed
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaProvider>
`;

exports[`should match the snapshot while loading 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {},
        Object {
          "dataSet": Object {},
        },
      ]
    }
    dataSet={Object {}}
    style={Object {}}
  >
    <View>
      <View
        dataSet={Object {}}
        style={
          Array [
            Object {},
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          dataSet={Object {}}
          fadeDuration={0.1}
          isLoaded={false}
          speed={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 3,
              "height": 80,
              "overflow": "hidden",
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#e5e5e5",
                "height": "100%",
                "opacity": 0,
                "width": "100%",
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={
            Object {
              "paddingBottom": 8,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 8,
            }
          }
        >
          <View
            _text={
              Object {
                "color": "lightText",
              }
            }
            accessible={true}
            collapsable={false}
            dataSet={Object {}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onHoverIn={[Function]}
            onHoverOut={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#24303E",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          >
            <Text
              dataSet={Object {}}
              level="1"
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#FFFFFF",
                  "fontFamily": "Roboto_400Regular",
                  "fontSize": 18,
                  "fontStyle": undefined,
                  "fontWeight": undefined,
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "textAlign": "auto",
                  "textDecorationLine": undefined,
                }
              }
            >
              name
            </Text>
            <Text
              dataSet={Object {}}
              level="1"
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#FFFFFF",
                  "fontFamily": "Roboto_400Regular",
                  "fontSize": 18,
                  "fontStyle": undefined,
                  "fontWeight": undefined,
                  "letterSpacing": 0,
                  "lineHeight": 26,
                  "textAlign": "auto",
                  "textDecorationLine": undefined,
                }
              }
            >
              closed
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaProvider>
`;

exports[`should match the snapshot with error 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {},
        Object {
          "dataSet": Object {},
        },
      ]
    }
    dataSet={Object {}}
    style={Object {}}
  >
    <View>
      <View
        dataSet={Object {}}
        style={
          Array [
            Object {},
            Array [
              Object {},
            ],
          ]
        }
      >
        <Text
          dataSet={Object {}}
          level="1"
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#CE3616",
              "fontFamily": "Roboto_500Medium",
              "fontSize": 32,
              "fontStyle": undefined,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 38,
              "textAlign": "auto",
              "textDecorationLine": undefined,
            }
          }
        >
          error
        </Text>
        <View
          dataSet={Object {}}
          style={
            Object {
              "paddingBottom": 8,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 8,
            }
          }
        >
          <View
            _text={
              Object {
                "color": "lightText",
              }
            }
            accessible={true}
            collapsable={false}
            dataSet={Object {}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onHoverIn={[Function]}
            onHoverOut={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#24303E",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              }
            }
          >
            <Text
              dataSet={Object {}}
              level="1"
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#FFFFFF",
                  "fontFamily": "Roboto_400Regular",
                  "fontSize": 18,
                  "fontStyle": undefined,
                  "fontWeight": undefined,
                  "letterSpacing": 0,
                  "lineHeight": 21,
                  "textAlign": "auto",
                  "textDecorationLine": undefined,
                }
              }
            >
              name
            </Text>
            <Text
              dataSet={Object {}}
              level="1"
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#FFFFFF",
                  "fontFamily": "Roboto_400Regular",
                  "fontSize": 18,
                  "fontStyle": undefined,
                  "fontWeight": undefined,
                  "letterSpacing": 0,
                  "lineHeight": 26,
                  "textAlign": "auto",
                  "textDecorationLine": undefined,
                }
              }
            >
              closed
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaProvider>
`;
